packages/state_queue/lib/src/state_queue.dart

```diff
@@ -94,6 +99,10 @@ abstract class StateQueue<T> extends ValueNotifier<T>
 
   final _taskQueue = StreamController<_QueueEntry<T>>();
 
+  final _pendingOperations = PendingOperations();
+
+  PendingOperationsReader get pendingOperations => _pendingOperations;
```

Comment:
As discussed, we might want to implement `runQueuedTasksToCompletion` using `pendingOperations.awaitEndOfKnownCalls`. Can also do that as a follow up.